catalogue_pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/services/Merger.scala

```diff
@@ -0,0 +1,59 @@
+package uk.ac.wellcome.platform.merger.services
+
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.models.work.internal._
+
+class Merger extends Logging {
+  def merge(works: Seq[UnidentifiedWork]) = {
+    mergePhysicalDigitalPair(works)
+      .getOrElse(works)
+  }
+
+  private def mergePhysicalDigitalPair(works: Seq[UnidentifiedWork]) = {
+    if (works.size == 2) {
+      val (digitalWorks, physicalWorks) = works.partition(isDigitalWork)
+      mergePhysicalAndDigitalWorks(physicalWorks, digitalWorks)
+    } else {
+      None
+    }
+  }
+
+  private def mergePhysicalAndDigitalWorks(
+    physicalWorks: Seq[UnidentifiedWork],
+    digitalWorks: Seq[UnidentifiedWork]) = {
+    if (physicalWorks.size == 1 && digitalWorks.size == 1) {
+      // As the works are supplied by the matcher these are trusted to refer to the same work without verification.
+      // However, it may be prudent to add extra checks before making the merge here.
+      val physicalWork = physicalWorks.head
+      val digitalWork = digitalWorks.head
+      Some(
+        List(
+          mergePhysicalWithDigitalWork(physicalWork, digitalWork),
+          redirectWork(digitalWork, physicalWork.sourceIdentifier)))
+    } else {
+      None
+    }
+  }
+
+  private def redirectWork(workToRedirect: UnidentifiedWork,
+                           redirectTargetSourceIdentifier: SourceIdentifier) = {
+    UnidentifiedRedirectedWork(
+      sourceIdentifier = workToRedirect.sourceIdentifier,
+      version = workToRedirect.version,
+      redirect = IdentifiableRedirect(redirectTargetSourceIdentifier)
+    )
+  }
+
+  private def isDigitalWork(work: UnidentifiedWork): Boolean = {
+    work.workType match {
+      case None    => false
+      case Some(t) => t.id == "v" && t.label == "E-books"
```

Comment:
Why do we have to check for id being "v"? Why is the check on label not enough?

Reply:
I have no idea, I think it's just the example I got.  Will check.

Reply:
Have confirmed this is correct at present given what Sierra holds, but will change in the future.